test/runtests.jl

```diff
@@ -1,315 +1,556 @@
+            @test size(Data.schema(data)) == (2,27)
+            ODBC.execute!(dsn, "drop table if exists test1")
+        end
+
+        @testset "large query" begin
+            ODBC.execute!(dsn, "drop table if exists test2")
+            ODBC.execute!(dsn, """
+            CREATE TABLE test2
+            (
+                ID INT NOT NULL PRIMARY KEY,
+                first_name VARCHAR(25),
+                last_name VARCHAR(25),
+                Salary DECIMAL,
+                `hourly rate` real,
+                hireDate DATE,
+                `last clockin` DATETIME
+            );""")
+            randoms = joinpath(dirname(@__FILE__), "randoms.csv")
+            ODBC.execute!(dsn, "load data local infile '$randoms' into table test2
+                                fields terminated by ',' lines terminated by '\n'
+                                (id,first_name,last_name,salary,`hourly rate`,hiredate,`last clockin`)")
+
+            data = ODBC.query(dsn, "select count(*) from test2")
+            @test size(Data.schema(data)) == (1,1)
+            @test data[1][1] === 70000
+
+            df = ODBC.query(dsn, "select * from test2")
+            @test size(Data.schema(df)) == (70000,7)
+            @test df[1] == collect(1:70000)
+            @test df[end][1] === ODBC.API.SQLTimestamp(2002,1,17,21,32,0,0)
+        end
+
+        @testset "prepared statement" begin
+            ODBC.execute!(dsn, "create table test3 as select * from test2 limit 0")
+            ODBC.execute!(dsn, "delete from test3")
+
+            stmt = ODBC.prepare(dsn, "insert into test3 values(?,?,?,?,?,?,?)")
+            ODBC.execute!(stmt, [101, "Steve", "McQueen", 1.0, 100.0, Date(2016,1,1), DateTime(2016,1,1)])
+
+            ODBC.execute!(stmt, [102, "Dean", "Martin", 1.5, 10.1, Date(2016,1,2), DateTime(2016,1,2)])
+
+            df = ODBC.query(dsn, "select * from test3")
+            @test size(Data.schema(df)) == (2,7)
+            @test df[1][end-1] == 101
+            @test df[1][end] == 102
+            @test df[2][end-1] == "Steve"
+            @test df[2][end] == "Dean"
+            @test df[3][end-1] == "McQueen"
+            @test df[3][end] == "Martin"
+            @test df[4][end-1] == DecFP.Dec64(1)
+            @test df[4][end] == DecFP.Dec64(2)
+            @test df[5][end-1] == 100.0
+            @test df[5][end] == 10.1
+            @test df[6][end-1] == ODBC.API.SQLDate(2016,1,1)
+            @test df[6][end] == ODBC.API.SQLDate(2016,1,2)
+            @test df[7][end-1] == ODBC.API.SQLTimestamp(2016,1,1,0,0,0,0)
+            @test df[7][end] == ODBC.API.SQLTimestamp(2016,1,2,0,0,0,0)
+            ODBC.Source(dsn, "drop table if exists test2")
+            ODBC.Source(dsn, "drop table if exists test3")
+        end
+
+        ODBC.disconnect!(dsn)
+        println("mysql passed.")
+    else
+        warn("Could not connect to mysql server: Skipping mysql tests")
+    end
+end
+
+@testset "postgresql" begin
```

Comment:
If theses various testsets don't overlap then I think it's sensible to put their contents in separate files. I would keep the "TEST_*" logic in the "runtests.jl" file.